src/Products/ZCatalog/Catalog.py

```diff
@@ -897,29 +886,83 @@ def _sort_nbest_reverse(self, actual_result_count, result, rs,
                         n += 1
                     best = keys[-1]
         else:
-            for did in rs:
+            # we have multi index sorting
+            result = self._multi_index_nbest(
+                actual_result_count,
+                result,
+                rs,
+                limit,
+                sort_index,
+                sort_spec,
+                second_indexes_key_map,
+                reverse=False
+            )
+
+        return (actual_result_count, 0, result)
+
+    def _multi_index_nbest(self, actual_result_count, result,
+                           rs, limit, sort_index, sort_spec,
+                           second_indexes_key_map, reverse=True):
+        """
+        For multiple indexes.
+        1) Categorize documents as lists by the first index values in the
+        did_by_index_value dict.
+        2) Sort the index_values
+        3) Collect from the did_by_index_value dict by the sorted
+        index_values till limit is exeeded.
+        """
+        # A dict of lists categorize the documents after the first sort
+        # index value.
+        did_by_index_value = {}
+        # get the index' keymap
+        index_key_map = sort_index.documentToKeyMap()
+        # for all documents
+        for did in rs:
+            # get the index value of the current document id
+            try:
+                index_value = index_key_map[did]
+            except KeyError:
+                # This document is not in the sort key index, skip it.
+                # ToDo: Is this the correct/intended behavior???
+                actual_result_count -= 1
```

Comment:
As `actual_result_count` is an `int` and thus immutable, this does not change the value in the caller. As mentioned before, you must do something special to get the updated value to the caller (e.g. return the updated value as part or the return value).